src/ray/common/ray_config_def.h

```diff
@@ -463,6 +463,11 @@ RAY_CONFIG(int64_t, task_events_max_num_task_events_in_buffer, 10000)
 /// Setting the value to -1 allows unlimited profile events to be sent.
 RAY_CONFIG(int64_t, task_events_max_num_profile_events_for_task, 100)
 
+/// The delay in ms that GCS should mark any running tasks from a job as failed.
+/// Setting this value too smaller might result in some finished tasks marked as failed by
+/// GCS.
+RAY_CONFIG(uint64_t, gcs_mark_task_failed_on_job_done_delay_ms, /*  3 secs */ 1000 * 3)
```

Comment:
Let's have a bit longer delay like 10 seconds, so we can have more time to wait for the report

Reply:
The wait time for the report should be pretty quick IIUC tho? As soon as the driver exits, the raylet will clean up any workers owned by the driver (any running tasks from the job will be killed) and GCS will have the job finished state reported?

Reply:
I think it is safer having a bit longer delay since 1 worker can have lots of tasks buffered, and it takes time to report all. 2. GCS could have been delayed. So I think 3 seconds are pretty short. I originally thought even 30 seconds, but it might be too long lol... I think 10 seconds are not too long and kind of reasonable delay

Reply:
I see, I thought a worker taking a few seconds to report is more of an edge case while most of the time, all workers should have been shutdown pretty quickly when the job is over, which is more of the common case that the default delay value should cater to. 

But I am open to keep this longer like 10 secs since it's tunable.

Reply:
Let's try 15 seconds (enough delay)!